test/e2e/performance_test.go

```diff
@@ -0,0 +1,371 @@
+	return err
+}
+
+// createPerformanceNginx creates the nginx pod and wait it to be ready.
+func createPerformanceNginx(data *TestData, b *testing.B) (string, error) {
+	b.Logf("Creating a nginx test Pod")
+	if err := createNginxPodOnMaster(data); err != nil {
+		b.Fatalf("Error when creating nginx test pod: %v", err)
+	}
+	b.Logf("Waiting IP assignment of the nginx test Pod")
+	return data.podWaitForIP(defaultTimeout, performanceNginxPodName)
+}
+
+// createPerformanceAB creates the ab pod and wait it to be ready.
+func createPerformanceAB(data *TestData, b *testing.B) (string, error) {
+	b.Logf("Creating an ab test Pod")
+	if err := createABPodOnMaster(data); err != nil {
+		b.Fatalf("Error when creating ab test pod: %v", err)
+	}
+	b.Logf("Waiting IP assignment of the ab test Pod")
+	return data.podWaitForIP(defaultTimeout, performanceABPodName)
+}
+
+// setupPerformanceTestPodsConnection applies the network policy which enable connectivity between test pods to the cluster.
+func setupPerformanceTestPodsConnection(data *TestData) error {
+	npSpec := networkv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{"app": performanceAppLabel}},
+		Ingress: []networkv1.NetworkPolicyIngressRule{
+			{
+				From: []networkv1.NetworkPolicyPeer{
+					{
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": performanceAppLabel},
+						},
+					},
+				},
+			},
+		},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	np := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: podsConnectionNetworkPolicyName},
+		Spec:       npSpec,
+	}
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func generateWorkload(amount int) *networkv1.NetworkPolicy {
+	var ingressRules []networkv1.NetworkPolicyPeer
+	rndSrc := rand.NewSource(seed)
+	existed := make(map[string]struct{}) // ensure no duplicated cidrs
+	for len(ingressRules) < amount {
+		cidr := randCidr(rndSrc)
+		if _, ok := existed[cidr]; ok {
+			continue
+		} else {
+			existed[cidr] = struct{}{}
+		}
+		ingressRules = append(ingressRules, networkv1.NetworkPolicyPeer{IPBlock: &networkv1.IPBlock{CIDR: cidr}})
+	}
+	npSpec := networkv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{"app": performanceAppLabel}},
+		Ingress:     []networkv1.NetworkPolicyIngressRule{{From: ingressRules}},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	return &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: workloadNetworkPolicyName},
+		Spec:       npSpec,
+	}
+}
+
+func populateWorkload(np *networkv1.NetworkPolicy, data *TestData) error {
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func setupPerformanceTestPods(data *TestData, b *testing.B) (nginxPodIP, abPodIP string) {
```

Comment:
adding "Performance" in all the names is a bit weird and perhaps unnecessary

Reply:
I still see "Performance" in all the names despite your reply?

Reply:
Ops, I missed it. But these tests are all under the same package, I think `setupTestPods` is too general to be distinguished from other tests?